Add timing.hpp and make it a backend of the tracing system (#575)
kpet/clvk: README.md

LD_LIBRARY_PATH=/path/to/build clinfo
```

### With global timing information

> Global timing information about API functions as well as some internal functions can be printed on stdout at the end of the execution by clvk.

Reviewer: ```suggestion
Global timing information about API functions as well as some internal functions can be logged at the end of the execution.
```
